### DISCUSSION TITLE
OpenCPN Manual: Future improvements

### DISCUSSION THREAD

**Discussion by rgleason** [opencpn-manuals/main#30] [General]
### Goals: Main Manual rewrite/update

1. Easy and quick to use and update.
2. Easy to learn.
3. Accessible to developers and users alike.
4. Maintained by Users (largely).
5. Balance security and open community access. Avoid spammers.
6. Screenshots: All new, abbreviated names based on menus, resolution standards, stored in one location.
7. Reorganize into two parts "Basic" (included with releases)  and "Advanced" (optional download added to "Basic").
8. This is going to be a big task, requiring User Volunteers, so we want to have it easy to learn.

### Main Discussion: 

Decide between Continuing to use Dokuwiki and moving to Antora/Ascii.
Note that all Plugin Manuals and the Dev Manual have been moved to Antora/Ascii, largely by Mike and Alec and that system seems to be working well, but is not as readily accessible to Users without github accounts.

#### Antora/Ascii Pros/Cons

**Comment by Rasbats** [comment]
>
> not as readily accessible to Users without github accounts
>

But the whole point of the plugins manual was to put the manual editing in the hands of the plugin developer. He/she can accept user comments via a number of pathways (Forum, GitHub, Email) and edit/correct the manual. At the next build of opencpn-manuals this work is included in the manual. The Source List (thanks Alec) shows the latest commit for the plugin, which will include that work.

**Reply by rgleason** [reply]
The  plugin developer has always had access and control of their manual with a little help from admins. The main point of the new plugins manual for me, is to clean up what was becoming a big messy nest so that we could get control of the documents, leaving the OpenCPN user manual to work with.  Quite frankly I find the antora/asciidoc process quite cumbersome and slow. It is not easy to edit without being able to see the result, and then edit again.  So I don't like it.

**Comment by Rasbats** [comment]
Agree with the concept of a basic user manual and an advanced manual.

#### Dokuwiki Pros/Cons

__Pros__

- Many users with experience of the system
- Images easily added

__Cons__

- No versioning system that allows the manual to be frozen at a particular point in time
- Each contributor is in effect an editor, leading to bloat and inconsistencies of style
- Location of image files inconsistent in the present Dokuwiki
- Difficult to create a stand alone website folder to act as a help file
- New authors may find the Dokuwiki editing difficult to learn

#### Antora/Ascii Pros/Cons

__Pros__

- A modern markup system that has the backing of a number of entities
- Nice presentation
- Antora creates a website folder that can be used as a stand alone help system. A template for this already exists in the Weatherfax plugin
- Many content editors (software) are available (including online)
- Any user can contribute work but a limited number of editors, ensuring consistent style and avoiding bloat
- Work could be divided up between authors and then incorporated in the manual by the editors.
- Work in the Dokuwiki can be translated to asciidoc and text/images etc used as a starting point for the new manuals.
- Versioning

__Cons__

- A learning curve
- A shared image repository is not yet available (Investigating Google Drive at present)
- Pathway for users to contribute to the master document needs developing

**Reply by rgleason** [reply]
I agree about the location of images in docuwiki. That is one of the things that needs to be tackled and enforced and clarified.
There is a plugin which allows moving the images and changing all related links, it is a bit cumbersome though.

With docuwiki, we could manually freeze the document, copy it, and resume editing. I have never done this, was have always just moved forward.  Versioning might be useful, but it is not used for plugins either.

I don't understand "Difficult to create a stand alone website folder to act as a help file"  How so?
We have the Editors Manual which could be improved.

Learning Dokuwiki is easier than learning asciidoc in my opinion. In both instances new users must learn, but the level of learning is greater and the barriers are greater with antora/asciidoc because they have to have a github account and at least understand something about commits.

Quite frankly, I find the Dev Manual, while it is an improvement,  navigating it is much worse than what we had with dokuwiki. There are inconsistencies in the use of the pages.  There are placemarkers for organization purposes, that have no page associated, so that navigation is not good.  There is also going to be duplication when Testplugin is fully done, because content is not shared.

**Comment by Rasbats** [comment]
>I don't understand "Difficult to create a stand alone website folder to act as a help file" How so?
We have the Editors Manual which could be improved.

I was thinking of a standalone basic help system for OpenCPN that did not require an Internet connection. The "Basics Manual", that should be available as part of the installation.

**Comment by Rasbats** [comment]
For WYSIWYG online editing I found this recently: [https://thetimetube.herokuapp.com/asciidoc/](https://thetimetube.herokuapp.com/asciidoc/) (F9 to get a preview of the page).

Visual Studio Code is easy to use. Ctrl-Shift-V for the preview.

**Comment by rgleason** [comment]
That manual is an html file that I made sometime ago, that Dave included as documentation. I also made several versions of epub.  Unfortunately it is outdated, thus the first effort should be to clean up and edit a "Basic Manual" I think.  The remaining would be in the Advanced Manual.  I have outlined these two manuals elsewhere.

**Comment by Rasbats** [comment]
For WYSIWYG online editing I found this recently: [https://thetimetube.herokuapp.com/asciidoc/](https://thetimetube.herokuapp.com/asciidoc/) (F9 to get a preview of the page).

Visual Studio Code is easy to use. Ctrl-Shift-V for the preview.

@rgleason : How about I produce a basics manual using asciidoc. I can cherry pick from work done last year. This will allow production of a pdf and website help folder. It can be independent of the user manual or incorporated later.

**Comment by rgleason** [comment]
Well, Mike, you are the leader on antora/asciidoc and I am the follower. You did a great job moving us out of a gordons knot with all the plugins. If you'd like to start this that would be great, but let's talk about it first and try to do some planning and agree on some things.

If we are going to move this into antora/asciidoc we are not going to have as much help from users. I think we will get some help, but not all user/editors have github accounts and the team will consequently be much smaller.

Let's discuss organizaton and indexes.  Please see the Sections that I've proposed for the [BASIC USER MANUAL](https://opencpn.org/wiki/dokuwiki/doku.php?id=opencpn:manual)   It is 11 sections that should look familiar to you. The[ ADVANCED USER MANUAL](https://opencpn.org/wiki/dokuwiki/doku.php?id=opencpn:manual_advanced) is about five main sections.  Please look through this and advise what you think.

Also are there some changes/additions that should be made to organization to consolidate and make it more comprehensible?
For example,  is there a way to improve Routes & Waypoints  and Create a Route?  Things get more complex with Autopilots and activating a route. These topics are all interrelated.  How do we make things simpler and clearer?  [We need to put on our good editors hat here!}

The other thing we need to do is create a good system for naming and locating all screenshots and establishing some standards for the size and format for screenshots. I think we should update/redo almost all of the screenshots, which is a specific task.

1. Naming Menu Screenshots: Use the Menu System but abbreviated.  Any suggestions?  Perhaps we should develop the full naming from the menu system first?

- Example: Toolbar= tb,  Options=op,  Display=dis. Connection=con, Ships=shp, User Interface=ui,  Plugins=pi 
- Example: Options=op  Tabs OwnShip=own  AIS Targets=ais, MMSI Properties=mmsi, Routes/Points=rts
- Example Screenshot names:  tb-op-shp+own,  tb-op-shp+ais,   tb-op-shp+mmsi, tb-op-shp+rts

2. Menu Screenshot Resolution  - I will have to look at this further, see[ these resolutions](https://opencpn.org/wiki/dokuwiki/doku.php?id=opencpn:opencpn_user_manual:getting_started:opencpn_installation&do=media&ns=opencpn%3Aopencpn_user_manual%3Agetting_started), but there are a number of sizes needed, about 500x500 for a full screen showing menus. and then smaller sizes for smaller menus.  

4. Other Screenshots - Also sometimes it is helpful to have a menu and a part of the screen showing some aspect or symbol, these may be bigger. These don't need to be as "standardized" but just suit the need.

5. Menu Screenshots Location - In one central place.

So I have run out of steam here, and probably should let you respond.

**Comment by rgleason** [comment]
RE:  For WYSIWYG online editing I found this recently: https://thetimetube.herokuapp.com/asciidoc/ (F9 to get a preview of the page).

Visual Studio Code is easy to use. Ctrl-Shift-V for the preview.

I could live with https://thetimetube.herokuapp.com/asciidoc/ (F9 to get a preview of the page).
but Wish they had a reference page for coding.
I hope you will give us some instructions on how you handle this now.

**Reply by Rasbats** [reply]
Better to continue this discussion on Zulip. I will start a stream for the Basics Manual.

**Reply by rgleason** [reply]
@Rasbats Rasbats, I think we should keep this discussion here, so others can participate, because the Zulip portal is pretty far down in the weeds and quite frankly, I don't think it has enough public eyes on it thinking about this.  If we have some nitty detail we are working out then maybe we should do it on Zulip and then come back here with the result?   --Is that what you intend?

**Reply by Rasbats** [reply]
@rgleason Zulip was only intended for the Basics Manual I am proposing.

**Comment by Rasbats** [comment]
https://asciidoc-py.github.io/manpage.html

Includes a cheatsheet.

**Reply by rgleason** [reply]
Mike thats good, I think the cheatsheet should be readily available through whatever interface/editor that we use. It should be short and concise.

Go  ahead use zulip but lets publish conclusions to the github discussions pleasr.

**Comment by Rasbats** [comment]
I would like to make the "Basics Manual" available in the major languages. I have been playing with po4a, which creates a .pot file. After translation the .po file is used to created a new .adoc file that can be used to build a new website page in the manual. Intention is to keep the manual as simple as possible to ease the burden of translation.

**Comment by rgleason** [comment]
Docuwiki also supports multiple languages, and has done so for some time

**Comment by Rasbats** [comment]
First draft of the Basics Manual is here:
[https://rasbats.github.io/opencpn-basics/basics_manual/0.1/index.html](https://rasbats.github.io/opencpn-basics/basics_manual/0.1/index.html)

**Comment by Rasbats** [comment]
Improve the navigation.
[https://rasbats.github.io/opencpn-basics/basics_manual/0.2/GPS/gps.html](https://rasbats.github.io/opencpn-basics/basics_manual/0.2/GPS/gps.html)

**Comment by rgleason** [comment]
Mike, looks good, but just a small thing in the title.  I think it should be OpenCPN Basic Manual  without the "s".

**Comment by rgleason** [comment]
Also this link does not seem to pop a message up
Hover the mouse over [HERE](https://opencpn.org/) and the tooltip shows where the link leads.

**Comment by Rasbats** [comment]
@rgleason : Thanks. It should read:
 Hover the mouse over [HERE](https://opencpn.org/) and the browser shows where the link leads.

Basic/Basics ... semantics. I use the phrase understanding the basics of OpenCPN. Maybe OpenCPN Essentials Manual is better? But what is essential?

**Comment by Rasbats** [comment]
[https://www.amazon.co.uk/Autodesk-Revit-Basics-Training-Manual/dp/0831136219](https://www.amazon.co.uk/Autodesk-Revit-Basics-Training-Manual/dp/0831136219)

**Comment by rgleason** [comment]
I think we should keep it simple and keep it short.  OpenCPN Basic Manual
There is a Basic Manual and an Advanced Manual.  It has nothing about plurality, the "s" is not needed.

**Comment by Rasbats** [comment]
@rgleason : You have the lead on this. I will change it to 'OpenCPN Basic Manual'.

**Reply by rgleason** [reply]
Thanks Mike.

**Comment by Rasbats** [comment]
Changes made:
[https://rasbats.github.io/opencpn-basic/basic_manual/0.3/index.html](https://rasbats.github.io/opencpn-basic/basic_manual/0.3/index.html)

**Comment by rgleason** [comment]
Mike, You've accomplished a lot here. The simplicity of this is very appealing and will make it easier for new users. Perhaps it will lead to more satisfied users who get the software installed properly and are happy to be using the program effectively.

What is your vision for how to handle all the details? In the old Manual we laboriously went through each and every setting, with a screenshot of the menu and then text describing the settings.  Is this really necessary?  What do we loose by not doing this?
Aren't the settings supposed to be self explanatory?

For example: Regarding Routes and the many new functions, which can get pretty detailed, under Basic perhaps just having the right click menu listed will be enough to get new users going?   Then under Advanced I expect that there will be some of those functions that will need more explanation. What do you think is the best way to handle this? 

Maybe if certain areas or items need more clarification we should add that? 
Or 
Perhaps as new users get more familiar, they will have questions and want to know more. Do those more detailed sections then become part of Advanced or should they be in Basic?

Another example is Setting up GPS  What we have in the old manual is the result of all the trouble that users were having, due to multiple OS, and many different GPS manufacturers, confusion about the use of a serial port, etc.  The current page looks good, but what happens when someone hits a roadblock?  Do we refer them to the old manual?   Or do we put that in GPS addendum page or have it in the Advanced Manual?

I am very glad you got this going!

**Comment by Rasbats** [comment]
@rgleason ...
>What is your vision for how to handle all the details?

Keep the Basic ... basic. The Advanced Manual is available if a new user wants to go further and explore other options. How about a page of references for subjects not covered. All in one place so we don't have a problem with updating pages. 

You will have noticed from the asciidoc that _:experimental:_ is used in the header. Not really experimental as it is functioning well. It includes the option, used in Basic, for a way of indicating a button. **btn:[Cancel]** for instance. This results in **[Cancel]** in the web page.
By using this markup, which uses text already translated for the main OpenCPN program, we could make translations easier. Press this or that is not needed.

When sailing allows I will move the Basic Manual repo to opencpn-manuals and use the template already started for Inland Ecdis.
[https://opencpn-manuals.github.io/inland-ecdis/manuals/en/index.html](https://opencpn-manuals.github.io/inland-ecdis/manuals/en/index.html)
For Inland Ecdis the docs are in a separate repo, separated from the repo that builds the website, which makes the docs easier to fork and change. Thanks @leamas .

**Reply by rgleason** [reply]
@Rasbats I think a page of links to topics not covered that goes to the Advanced Manual is good.  

Thinking about keeping things simple, but allowing users to drill down for further detail..  Is having such "more detail" links in basic which are located appropriately at the end of a topic ok?

**Comment by rgleason** [comment]
@Rasbats I sent you via email and onedrive link a 35mb ppt power point about using opencpn. It is from one users perspective, but it may provide us with some addirional basic topics that should be covered. I will also compare later.

**Comment by rgleason** [comment]
Where is the Basic manual now?  It has moved.

**Comment by rgleason** [comment]
I see how the inland ecdis works for multiple language. This looks very good to me!  Each language has a separate directory. I expect that the en manual is copied to a new language and the translator simply types over and edits, then commits. I would be interested in what you have found out.

**Comment by Rasbats** [comment]
All ...

The OpenCPN Basic Manual is now in the opencpn-manuals repositories: 
[https://github.com/opencpn-manuals/basic](https://github.com/opencpn-manuals/basic
)
First build of the manual using this repo:
[https://opencpn-manuals.github.io/basic/manuals/en/](https://opencpn-manuals.github.io/basic/manuals/en/)

I am now going to archive Rasbats/opencpn-basic.

**Comment by Rasbats** [comment]
> I would be interested in what you have found out.

See the previous post. po4a can be used to create a .pot file from the asciidoc for translating with poedit. po4a is a package that can be added to cygwin.

**Reply by rgleason** [reply]
Sorry I don't find that post Mike, but very interested. BTW  I've tried to log into Zulip, but there is some problem, probably associated with my moving from a very old failing computer to this new laptop.  I will try to get on zulip today.

**Comment by rgleason** [comment]
`Keep the Basic ... basic. The Advanced Manual is available if a new user wants to go further and explore other options. How about a page of references for subjects not covered. All in one place so we don't have a problem with updating pages.`

@Rasbats 
Is a link within a page acceptable with regard to updating links?
Is a link within other pages to a Reference page acceptable? 
If so, since all the "Advanced -Detail" links are listed in the "Page of References" why can't we have a link?

**This is a major usability issue, and is very different than Docuwiki and it needs to be solved.**

What exactly is the problem with updating links and how does putting a bunch of links on one page solve that?
You just don't get errors sprinkled throughout the pages, so they can all be solved on that one page.

What do you loose by doing this? User continuity. The link should be associate with the topic and located in the document where it needs to be located!  This is a "USER MANUAL" not a Dev manual.  

Also there should be tools to help identify and fix bad links to make editing easier.

**Reply by Rasbats** [reply]
>Is a link within other pages to a Reference page acceptable?

Why not use the navigation bar on the left? I have now added it.

>Is a link within other pages to a Reference page acceptable?

IMO No. It means each page with a link needs changing if the link changes. All on one page makes life easy.

>What exactly is the problem with updating links and how does putting a bunch of links on one page solve that?
You just don't get errors sprinkled throughout the pages, so they can all be solved on that one page.

I think you have answered your own question. One page of links in various sections for each topic in the manual.

Now I am off to enjoy the sunshine.

**Comment by rgleason** [comment]
And so you should!  Sunshine is needed.

Mike, you may have a different perspective. I am interested primarily in User Comprehension.

If asciicdoc/antora does not have tools like Docuwiki  multiOrphan to chase bad links, that is no reason to sacrifice User comprehension to a cumbersome structure that satisfies some artificial restriction about linkages to make it easier for Editors!

After the Basic section is written and reviewed, we can add the necessary Basic links to the appropriate lines. Things will be relatively settled at that point after the majority of organizational decisions have been made.

What is the best way to organize Basic and Advanced with respect to the asciidoc/Antora structure? The reason I ask is it may be easier to build the structure in the form we want at the end.  Should the two section be

```
OpenCPN Manual
   Basic
   Advanced
``` 
such that  nav.doc has something like this?

```
Basic Manual
[partial$nav-basic.adoc](https://github.com/opencpn-manuals/plugins/blob/main/manual/../modules/ROOT/partial$nav-basic.adoc)
Advanced Manual
[partial$nav-advanced.adoc](https://github.com/opencpn-manuals/plugins/blob/main/manual/.../modules/ROOT/partial$nav-advanced.adoc)

```

Obviously we need to be able to make a PDF/html of each section but this normally readily accomplished, no mater what structure. 

It would be much preferable to have Basic and Advanced built together in the same nav.adoc structure so that links between the two sections can be made, simply using the the index outline structure.  We do not want a firewall build between Basic and Advanced that would be a structure that should be avoided at all costs.

By the look of it that is the direction being taken now.

**Reply by Rasbats** [reply]
>If asciicdoc/antora does not have tools like Docuwiki multiOrphan to chase bad links

The build machinery that Alec made, which is already in use for Basic, includes link checking.

> We do not want a firewall build between Basic and Advanced

The scenario, as I see it, is the same as we encountered with the plugins manual. That was developed as a standalone and then added:
[https://opencpn-manuals.github.io/main/manuals/index.html](https://opencpn-manuals.github.io/main/manuals/index.html)

Whatever route you take the new Advanced Manual can be added to the list of manuals.

You have full access to the opencpn-manuals/basic and basic-docs repos.

The Basic Manual needs translations, which is easier for that manual than the Advanced Manual.

**Comment by rgleason** [comment]
It does not include the tools available in multiOrphan which help to fix the errors.
The structure used for plugins is definitely not suitable for the OpenCPN User Manual, because links between different plugins is difficult and not encouraged.

The Basic Manual and Advanced manual should be the same document, due to the limited asciidoc/antora tools and scripts available and the attitude taken about links. Alex has made a link checker that **_bans all links between separate documents, which is an arbitrary decision that runs counter to having an easily used USER MANUAL_** because the user will not be able to drill down into details when needed.

This area is for discussion about "Future Improvements" to the User Manual. You and Alec have not participated in any discussion, but just decided to aggressively start to use asciidoc/antora without bothering to get the issues I am raising adequately resolved.  I do not see this as a cooperative environment. The asciidoc/antora   Basic document (after considerable time and effort) was just dumped on us without having important discussion.

I am still not convinced that asciidoc/antora is appropriate for the User Manual, and will maintain great reservations until the basic questions I have raised about links are properly answered.  I have some ideas about how to achieve this, but am not going to discuss them until we truly have a cooperative environment where ideas can be discussed reasonably.

I will be pursuing a separate path using Dokuwiki for these USER documents, if these issues are not adequately answered and demonstrated in these trials. Dokuwiki supports multiple languages and using "standalone" Basic and Advanced documents don't really address user needs unless we can be more flexible about the Basic Manual and make it as I originally envisioned, such that it contains everything that is needed for the topics that I listed originally, and it is not a totally stripped down shell that does not answer the necessary questions the users will inevitably have.  I have some ideas how to do this with asciidoc/antora but they will not work with the limitations on links that Alec has imposed.  Given this situation, I will pursue the Dokuwiki solution.

![Future-User-Manual](https://user-images.githubusercontent.com/2363959/178568686-eebd344a-7a39-4777-816e-1d0f22cb2788.png)

**Comment by leamas** [comment]
>  Alex has made a link checker that bans all links between separate documents, 

I'm still Alec. And there are no such rules I have been enforcing (Antora certainly doesn't). But, I just don't have time now.

**Comment by rgleason** [comment]
Alex your script has such restrictions, nor do I have time.

**Comment by leamas** [comment]
I'm still Alec. And scripts could be patched.
